03-rnaseq/dimension_reduction_rnaseq_01_pca.Rmd

```diff
@@ -0,0 +1,368 @@
+  "SRP133573.tsv" # Replace with the name of your data file
+)) %>%
+  # Here we are going to store the gene IDs as rownames so that we can have a numeric matrix to perform calculations on later
+  tibble::column_to_rownames("Gene")
+```
+
+Let's ensure that the metadata and data are in the same sample order. 
+
+```{r}
+# Make the data in the order of the metadata
+df <- df %>% dplyr::select(metadata$refinebio_accession_code)
+
+# Check if this is in the same order
+all.equal(colnames(df), metadata$refinebio_accession_code)
+```
+
+<!--ALL THE ANALYSIS STEPS HERE -->
+
+Now we are going to use a combination of functions from the `DESeq2` and `ggplot2` packages to look at the differential expression of our pre-ADT and post-ADT samples.
+
+## Create a DESeqDataset
+
+We will be using the `DESeq2` package for normalization, which requires us to format our data into a `DESeqDataSet` object.
+We turn the data frame (or matrix) into a [`DESeqDataSet` object](TODO: Add link to header section) and specify which variable labels our experimental groups using the [`design` argument](http://bioconductor.org/packages/devel/bioc/vignettes/DESeq2/inst/doc/DESeq2.html#multi-factor-designs) [@DESeq2].
+In this chunk of code, we will not provide a specific model to the `design` argument because we are not performing a differential expression analysis.
+
+```{r}
+# The `DESeqDataSetFromMatrix()` function needs the values to be converted to integers
+df <- df %>%
+  # Mutate numeric variables to be integers
+  dplyr::mutate_if(is.numeric, round)
+
+# We need to also format the variables from the metadata, that we will be using for annotation of the PCA plot, into factors
+metadata$refinebio_treatment <- as.factor(metadata$refinebio_treatment)
+metadata$disease <- as.factor(metadata$refinebio_disease)
+
+# Create a `DESeqDataSet` object
+dds <- DESeqDataSetFromMatrix(
+  countData = df, # This is the data.frame with the counts values for all replicates in our dataset
+  colData = metadata, # This is the data.frame with the annotation data for the replicates in the counts data.frame
+  design = ~1 # Here we are not specifying a model -- Replace with an appropriate design variable for your analysis
+)
+```
+
+## Define a minimum counts cutoff
+
+We want to filter out the genes that have not been expressed or that have low expression counts because we want to remove any possible noise from our data before we normalize the data and create our plot.
+We are going to do some pre-filtering to keep only genes with 10 or more reads total.
+Note that rows represent gene data and the columns represent sample data in our dataset.
+
+```{r}
+# Let's take a look at the distribution of counts first
+
+# Sum up the counts for each gene -- which means we want the sum of counts for each row as the rows represent gene data and the columns represent sample data in our dataset
+count_sums <- data.frame(count = rowSums(counts(dds)))
+
+# Create a density plot using the sums
```

Comment:
I know I said I thought this was a good idea to plot this out, but with how peak-y this plot is, I don't think it tells us much. Can we remove it from here (and maybe from clustering heatmap too). Sorry to change my mind on it. I usually do like plotting things before making cutoffs, but in execution I don't think it actually helps us. Sorry to double back!